packages/@aws-cdk/aws-sns-subscriptions/lib/url.ts

```diff
@@ -23,17 +23,17 @@ export interface UrlSubscriptionProps extends SubscriptionProps {
  * @see https://docs.aws.amazon.com/sns/latest/dg/sns-http-https-endpoint-as-subscriber.html
  */
 export class UrlSubscription implements sns.ITopicSubscription {
-  constructor(private readonly url: string, private readonly props: UrlSubscriptionProps = {}) {
-    if (!url.startsWith('http://') && !url.startsWith('https://')) {
+  constructor(private readonly url: string, private readonly protocol: sns.SubscriptionProtocol, private readonly props: UrlSubscriptionProps = {}) {
```

Comment:
I'd rather have the protocol optionally inside the props, and derive it from the literal URL (if available).

Only force passing it if the URL is unresolved.

Reply:
@rix0rrr fixed in d524d12